src/Void.Core/BufferMessages.fs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Void.Core
+
+type BufferMessage = inherit Message
+
+type BufferEnvelopeMessage<'TBufferMessage when 'TBufferMessage :> BufferMessage> =
+    {
+        BufferId : int
+        Message : 'TBufferMessage
+    }
+    interface EnvelopeMessage<'TBufferMessage>

```

Comment:
This is the first concretion on `EnvelopeMessage` that I'm seeing. I'm curious to see how it's useful.